TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,591 @@
+                    } else {
+                        encodersCanRun = true;
+                        //resets the encoders to a neutral value
+                        goalEncoderPosition = startEncoderValue;
+                        stop();
+                        return encodersCanRun;
+                    }
+                    break;
+                case SW:
+                    break;
+                case TURNLEFT:
+                    break;
+                case TURNRIGHT:
+                    if (getLeftEncoder() < goalLeftPosition) {
+                        drive(direction, power);
+                        telemetry.addData("Current Combined Value", combinedEnValue);
+                        telemetry.addData("Goal Value", goalEncoderPosition);
+                    } else {
+                        encodersCanRun = true;
+                        //resets the encoders to a neutral value
+                        goalEncoderPosition = startEncoderValue;
+                        stop();
+                        return encodersCanRun;
+                    }
+                    break;
+            }
+            return encodersCanRun;
+        }
+    }
+
+    //////////////////////////////////// GYRO
+
+    public boolean gyroTurn(Direction direction, double power, float degrees){
+        switch(direction) {
+            case TURNLEFT:
+                if (goalDegrees == -1) {
+                    goalDegrees = (this.getYaw() - degrees);
+                    if (goalDegrees < 0) {
+                        goalDegrees = (goalDegrees + 360);
+                    }
+                }
+                if (!(this.getYaw() > (goalDegrees - gyroRange) && this.getYaw() < (goalDegrees + gyroRange))) {
+                    drive(direction, power);
+                    telemetry.addData("Goal Degrees", goalDegrees);
+                    telemetry.addData("Current Degrees", getYaw());
+                    return false;
+                } else {
+                    this.stop();
+                    goalDegrees = -1;
+                    return true;
+                }
+            case TURNRIGHT:
+                if (goalDegrees == -1) {
+                    goalDegrees = (this.getYaw() + degrees);
+                    if (goalDegrees > 360) {
+                        goalDegrees = (goalDegrees - 360);
+                    }
+                }
+                if (!(this.getYaw() > (goalDegrees - gyroRange) && this.getYaw() < (goalDegrees + gyroRange))) {
+                    drive(direction, power);
+                    telemetry.addData("Goal Degrees", goalDegrees);
+                    telemetry.addData("Current Degrees", getYaw());
+                    return false;
+                } else {
+                    this.stop();
+                    goalDegrees = -1;
+                    return true;
+                }
+        }
+        return false;
+    }
+
+    public float getYaw(){
+        if(this.gyro.getYaw() < 0) {
```

Comment:
#fix `if (this.gyro.getYaw() < 0) {` and save `this.gyro.getYaw()` to a variable